nanoFramework.WebServer/HttpMultipartParser/LineBuffer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+
+namespace nanoFramework.WebServer.HttpMultipartParser
+{
+    internal class LineBuffer : IDisposable
+    {
+        private readonly Hashtable _data = new();
+
+        public void Dispose() => _data.Clear();
+
+        public void Write(byte[] bytes, int offset, int count)
```

Comment:
why not using a SpanByte instead of the byte array, the offset and the length?
you directly have functions to get and array, adding then to an ArrayList of hastable should do the trick perfectly.

Reply:
If I understand correctly the SpanByte would span a section of the buffer. The linebuffer contains the data where we are missing the \n. The newline will be in a chunk of data we haven't read yet. The buffer will be reused to read that next chunk of data and only when the next newline is received the call to LineBuffer.ToArray will follow. A SpanByte would hence loose the data as it would still be pointing to the same array (the buffer) whos data has changed in the meantime. (See also other comment about the mechanics)

Reply:
yes, that's the second I was thinking and get the buffer from SpanByte.ToArray() as the logic is already there. It should make us gain some buffer allocation if SpanByte are used from end to end.
But it would need a bit of investigation there as you're the one having the overall view on this.

Reply:
SpanByte works as expected and was committed. UnitTests are ok with it as well as a test with a file of 30kB

Reply:
Great :-) It will same an allocation every time. And it's used a lot in various bindings, with a native implementation. It's the Span<byte> but not generic as it's used in .NET IoT heavily.